src/pyhf/cli/contrib.py

```diff
@@ -0,0 +1,77 @@
+"""CLI for functionality that will get migrated out eventually."""
+import logging
+
+import click
+from urllib.parse import urlparse
+import tarfile
+from io import BytesIO
+from pathlib import Path
+
+try:
+    import requests
+except ModuleNotFoundError:
+    pass
```

Comment:
@kratsg As we don't have any tests that run the CLI with just a `pip install pyhf` these lines are going to be uncovered (and as such I've just let this import be in the try/except block instead of the whole function). Thoughts on this? Do we just eat these two lines of coverage? Or do we add a workflow that installs just the library defaults and then see if it is possible to patch coverage reports together?

Reply:
I might need an example of how you're suggesting to monkeypatch out `requests` from the environment. I haven't been able to find a good example of how to remove an entire module, but rather just attributes or items.

Reply:
@kratsg https://github.com/scikit-hep/pyhf/blob/e55eea408d7c28e3109338de96252119ac63f87a/tests/test_init.py#L31-L55 obviously well enough for removing different modules from `pyhf` but it still isn't clear to me how I would implement a `ModuleNotFoundError` that a user would see if they were missing `requests` and how to delete the library from view as to even create this error. Maybe I should follow up with you on this later though.

Reply:
Thanks @kratsg. Your example helped me out and made more sense after I had dinner. It is a slick solution that works very well for how simple it is. :)